app/models/dictionary.rb

```diff
@@ -293,15 +289,10 @@ def add_entries(raw_entries, analyzer)
     end
   end
 
-  def batch_normalize_entries(labels, analyzer)
-    norm1s = analyzer.batch_normalize(labels, normalizer1)
-    norm2s = analyzer.batch_normalize(labels, normalizer2)
-    [norm1s, norm2s]
-  rescue => e
-    raise ArgumentError, "Entries are rejected: #{e.message} #{e.backtrace.join("\n")}."
-  end
-
-  def new_entry(label, identifier, mode = EntryMode::GRAY, dirty = false, analyzer = Analyzer.new)
+  def new_entry(label, identifier)
+    mode = EntryMode::WHITE
+    dirty = true
+    analyzer = Analyzer.new
     norm1 = normalize1(label, analyzer)
     norm2 = normalize2(label, analyzer)
     Entry.new(label:label, identifier:identifier, norm1:norm1, norm2:norm2, label_length:label.length, mode:mode, dirty:dirty, dictionary_id: self.id)
```

Comment:
modeとdirtyはここでしか使わないので、展開して、次のように書けそうです。

```suggestion
    entries.build label: label,
                         identifier: identifier,
                         norm1: norm1,
                         norm2: norm2,
                         label_length: label.length,
                         mode: EntryMode::WHITE,
                         dirty: true
```

Dictionary IDの渡し方も変えてあります。
あと、引数が長いので改行してあります。

GitHub上でインデントが上手く調整できなかったので、ズレています。

Reply:
修正しました。
https://github.com/pubannotation/pubdictionaries/pull/160/commits/bca1a516348261f7c0c7eb425559960417b98260